liquigraph-core/src/test/java/org/liquigraph/core/configuration/ConnectionConfigurationByDataSourceTest.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.liquigraph.core.configuration;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Supplier;
+import org.junit.Test;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class ConnectionConfigurationByDataSourceTest {
```

Comment:
You can add a test where `dataSource.getConnection()` throws a `SQLException` to add coverage on the `catch` clause, asserting that the cause of the `RuntimeException` is the `SQLException`.

On a related note, I try these days to limit my use of `Mockito::verify` to testing methods with side effects. For tests like the ones below, I'll set up the mock to return some explicit value, and assert  that `connectionProvider.get()` returns it.